Refactor CMake with JRL CMake Modules v2

ahoarau opened the pull request:
⚠️DO NOT MERGE UNTIL https://github.com/jrl-umi3218/jrl-cmakemodules/pull/798 is merged ⚠️

This PR is a full rewrite of the CMake files with the [JRL CMake Modules v2](https://github.com/jrl-umi3218/jrl-cmakemodules/pull/798).

* Full rewrite of the CMake Files in modern CMake
* Remove submodules: archives can now be used
* `c++17` minimum required
* reorder tests to separate python and cpp tests
* 🚧 NIX CI is temporarily disabled

ahoarau committed:
remove duplicated EIGENPY_STRINGIZE macro
original is from templates/deprecated.hpp.in

ahoarau committed:
remove duplicated EIGENPY_STRINGIZE macro
original is from templates/deprecated.hpp.in

ahoarau committed:
update pre-commit and switch to gersemi

ahoarau committed:
update pre-commit and switch to gersemi

ahoarau committed:
update clang-format to c++17

ahoarau committed:
update clang-format to c++17

ahoarau committed:
fwd.hpp: remove space in user-defined litteral
deprecated https://en.cppreference.com/w/cpp/language/user_literal.html

ahoarau committed:
fwd.hpp: remove space in user-defined litteral
deprecated https://en.cppreference.com/w/cpp/language/user_literal.html

ahoarau committed:
ignore more stuff

ahoarau committed:
ignore more stuff

ahoarau committed:
add gersemi config

ahoarau committed:
add gersemi config

ahoarau committed:
remove submodules

ahoarau committed:
remove submodules

ahoarau committed:
rename main.cpp to bindings.cpp

ahoarau committed:
rename main.cpp to bindings.cpp

ahoarau committed:
remove custom python init file
they are now autogenerated by jrl v2

ahoarau committed:
remove custom python init file
they are now autogenerated by jrl v2

ahoarau committed:
update pixi.toml tasks

ahoarau committed:
update pixi.toml tasks

ahoarau committed:
separate cpp and python tests,  and generate variants
configure_file will overrite boost_optional.cpp only if the content changes. This allow to a quick check on the actual code, and simplifies the cmake

ahoarau committed:
separate cpp and python tests,  and generate variants
configure_file will overrite boost_optional.cpp only if the content changes. This allow to a quick check on the actual code, and simplifies the cmake

ahoarau committed:
rewrite CMake files with JRL CMake Modules v2

ahoarau committed:
rewrite CMake files with JRL CMake Modules v2

ahoarau committed:
intregrate get-jrl-cmakemodules.cmake

ahoarau committed:
intregrate get-jrl-cmakemodules.cmake

ahoarau committed:
update jrl url to point to fork

ahoarau committed:
update jrl url to point to fork

ahoarau committed:
require jrl-cmakemodules version 2.0.0

ahoarau committed:
require jrl-cmakemodules version 2.0.0

ahoarau committed:
include stubs pyi files in the installation

ahoarau committed:
include stubs pyi files in the installation

ahoarau committed:
ros-ci: add verbosity

ahoarau committed:
ros-ci: add verbosity

ahoarau committed:
changelog: add jrl v2 entry

ahoarau committed:
changelog: add jrl v2 entry

ahoarau committed:
add Copyright 2025-2026 Inria

ahoarau committed:
add Copyright 2025-2026 Inria

ahoarau committed:
remove the boost 1.90 restriction

ahoarau committed:
remove the boost 1.90 restriction

ahoarau committed:
remove unused options

ahoarau committed:
remove unused options

ahoarau committed:
update get-jrl-cmakemodules to display debug info

Check if v2 folder is present

ahoarau committed:
update get-jrl-cmakemodules to display debug info

Check if v2 folder is present

ahoarau committed:
ci: upgrade ci workflows
linux: tests and installs, uses cache
others: minor cosmetic changes

ahoarau committed:
ci: upgrade ci workflows
linux: tests and installs, uses cache
others: minor cosmetic changes

ahoarau committed:
put back the CNRS copyright

ahoarau committed:
put back the CNRS copyright

ahoarau committed:
unittests: fix pythonpath

ahoarau committed:
unittests: fix pythonpath

ahoarau committed:
ci: use ${{ github.workspace }}

ahoarau committed:
ci: use ${{ github.workspace }}

ahoarau committed:
c: add check configs

ahoarau committed:
c: add check configs

ahoarau committed:
ci: change default path

ahoarau committed:
ci: change default path

jorisv commented on unittest/cpp/CMakeLists.txt:
```diff
@@ -0,0 +1,75 @@
+
+set(TEST_TYPE "boost::optional")
+set(MODNAME boost_optional)
+configure_file(optional.cpp.in ${CMAKE_CURRENT_SOURCE_DIR}/boost_optional.cpp)
```
I'm not a huge fan of generating file in the source directory. It can be commited by mistake.
Maybe allow eigenpy_add_test to have a SOURCE_FILE optional argument that can take absolute path source file as argument.

ahoarau replied:
It was intentional as `configure_file` checks for the content before writing the file. I reverted to generating in the build dir. That's OK as well.

jorisv commented on unittest/cpp/boost_optional.cpp:
commited by mistake

jorisv commented on unittest/cpp/.clang-format-ignore:
To remove when generated in binary dir

jorisv commented on unittest/cpp/boost_variant.cpp:
commited by mistake

jorisv commented on unittest/cpp/std_optional.cpp:
commited by mistake

jorisv commented on unittest/cpp/std_variant.cpp:
commited by mistake

jorisv commented on unittest/cpp/test_boost_optional.py:
Commit by mistake ? I don't see where test_optional.py.in is used.

jorisv commented on unittest/cpp/test_boost_variant.py:
Commited by mistake ?

jorisv commented on unittest/cpp/test_optional.py.in:
Where is it used ?

jorisv commented on unittest/cpp/test_std_optional.py:
Commited by mistake ?

jorisv commented on unittest/cpp/test_std_variant.py:
Commited by mistake ?

jorisv commented on pixi.toml:
```diff
@@ -89,19 +88,39 @@ configure = { cmd = [
-  "-DPYTHON_STANDARD_LAYOUT=OFF",
+  "-DJRL_CMAKEMODULES_SOURCE_DIR=$JRL_CMAKEMODULES_SOURCE_DIR",
+  "--profiling-output=cmake-profiling.json",
+  "--profiling-format=google-trace",
```
To remove before merge

jorisv commented on CMakeLists.txt:
```diff
@@ -1,479 +1,317 @@
+jrl_configure_defaults()
 
-option(BUILD_WITH_CHOLMOD_SUPPORT "Build EigenPy with the Cholmod support" OFF)
+jrl_option(
+  GENERATE_PYTHON_STUBS
+  "Generate the Python stubs associated to the Python library"
+  OFF
+)
```
This option is not used. Stub are always generated.
This is an issue when we cross build (stub as to be imported, but it's impossible when we are cross building)

jorisv commented on CMakeLists.txt:
Prefix all options with EIGENPY_ when jrl_option support back compatibility.

ahoarau replied:
Let's do that in another PR.

jorisv commented on CMakeLists.txt:
```diff
@@ -1,479 +1,317 @@
-export_boost_default_options()
-find_package(Boost REQUIRED)
-search_for_boost_python(REQUIRED)
+jrl_find_package(Python 3.8 REQUIRED COMPONENTS Interpreter Development.Module NumPy)
```
Why you don't use `jrl_find_python` ?

jorisv commented on unittest/cpp/CMakeLists.txt:
```diff
@@ -0,0 +1,75 @@
+    TEST ${test_name}
+    PROPERTY
+      ENVIRONMENT_MODIFICATION
+        "PYTHONPATH=path_list_prepend:$<TARGET_FILE_DIR:eigenpy>/.."
```
I don't think this one is useful.
This give the `build/lib/..` directory where no imported lib can be found.

jorisv commented on CMakeLists.txt:
```diff
@@ -1,479 +1,317 @@
+  include/eigenpy/decompositions/SVDBase.hpp
+  include/eigenpy/decompositions/BDCSVD.hpp
+  include/eigenpy/decompositions/JacobiSVD.hpp
+  include/eigenpy/decompositions/minres.hpp
```
```suggestion
  include/eigenpy/decompositions/minres.hpp
  # Eigen
  include/eigenpy/eigen/EigenBase.hpp
```

jorisv commented on CMakeLists.txt:
Add ament support before we allow to merge.

ahoarau replied:
<img width="1259" height="144" alt="image" src="https://github.com/user-attachments/assets/14f69f6b-c8f5-4653-aa0b-fc103a43cfba" />

jorisv reviewed (changes requested):
Nice work, the CMakeLists are way clearer.

ahoarau committed:
pixi.toml: remove cmake profiling

ahoarau committed:
pixi.toml: remove cmake profiling

ahoarau committed:
ci: enable ros ci cache and prerelease

ahoarau committed:
ci: enable ros ci cache and prerelease

ahoarau committed:
rosci: configure cache

ahoarau committed:
rosci: configure cache

ahoarau committed:
cmake: set min to 3.22

ahoarau committed:
cmake: set min to 3.22

ahoarau committed:
cmake: use jrl_find_python

ahoarau committed:
cmake: use jrl_find_python

ahoarau committed:
cmake: generate stubs with GENERATE_PYTHON_STUBS condition

ahoarau committed:
cmake: generate stubs with GENERATE_PYTHON_STUBS condition

ahoarau committed:
cmake: install missing EigenBase header

ahoarau committed:
cmake: install missing EigenBase header

ahoarau committed:
ci: disable nix tmp

ahoarau committed:
ci: disable nix tmp

ahoarau committed:
gersemi: add back specific definitions

ahoarau committed:
gersemi: add back specific definitions

ahoarau committed:
cmake: remove extra pythonpath

ahoarau committed:
cmake: remove extra pythonpath

ahoarau committed:
cmake: reformat with updated gersemirc

ahoarau committed:
cmake: reformat with updated gersemirc

ahoarau committed:
ci: disable pixi explicit cache

ahoarau committed:
ci: disable pixi explicit cache

ahoarau committed:
ci: build verbose and limit to 2 cpu

ahoarau committed:
ci: build verbose and limit to 2 cpu

ahoarau committed:
ci: relax pixi cache

ahoarau committed:
ci: relax pixi cache

ahoarau committed:
ci: enable testing

ahoarau committed:
ci: enable testing

ahoarau committed:
ci: build in debug

ahoarau committed:
ci: build in debug

ahoarau committed:
ci: update cache keys

ahoarau committed:
ci: update cache keys

ahoarau committed:
cmake: replace jrl_find_package with jrl_check_python_module for scipy

ahoarau committed:
cmake: replace jrl_find_package with jrl_check_python_module for scipy

ahoarau committed:
generate generated tests in the build folder

ahoarau committed:
generate generated tests in the build folder

ahoarau committed:
remove clang-format ignore

ahoarau committed:
remove clang-format ignore

ahoarau committed:
cmake: update jrl_boostpy_add_stubs api

ahoarau committed:
cmake: update jrl_boostpy_add_stubs api

ahoarau committed:
cmake: add jrl_generate_ros2_package_files

ahoarau committed:
cmake: add jrl_generate_ros2_package_files

ahoarau committed:
cmake: generate ros2 python package files with condiition

ahoarau committed:
cmake: generate ros2 python package files with condiition

ahoarau committed:
ci: apt update before install

ahoarau committed:
ci: apt update before install

ahoarau committed:
ci: apt update before install

ahoarau committed:
ci: apt update before install

ahoarau committed:
rosci: do not use prerelease

ahoarau committed:
rosci: do not use prerelease

ahoarau committed:
cmake: update the get-jrl messages

ahoarau committed:
cmake: update the get-jrl messages

ahoarau committed:
cmake: require jrl >=1.2 to get the jrl-v2

ahoarau committed:
cmake: require jrl >=1.2 to get the jrl-v2

ahoarau committed:
cmake: update python and nanobind min required

ahoarau committed:
cmake: update python and nanobind min required

ahoarau committed:
pixi: update min python version

ahoarau committed:
pixi: update min python version

ahoarau committed:
pixi: add cxx-compiler to pixi-build feature

ahoarau committed:
pixi: add cxx-compiler to pixi-build feature

ahoarau committed:
cmake: include project version in config header generation

ahoarau committed:
cmake: include project version in config header generation

ahoarau committed:
pixi: update lock file

ahoarau committed:
pre-commit: change to gersemi precommit repo

ahoarau committed:
pre-commit: change to gersemi precommit repo

jorisv commented on CMakeLists.txt:
```diff
@@ -1,479 +1,317 @@
+jrl_find_package(Boost CONFIG REQUIRED COMPONENTS python numpy)
 
 if(BUILD_WITH_CHOLMOD_SUPPORT)
-  set(CMAKE_MODULE_PATH ${JRL_CMAKE_MODULES}/find-external/CHOLMOD
-                        ${CMAKE_MODULE_PATH})
```
CHOLMODConfig.cmake is not packaged on Ubuntu 22.04 (libsuitesparse-dev).
We must provide it to avoid regressions.
To avoid using our custom CHOMODConfig.cmake on all system we can use the same strategy than [aligator](https://github.com/Simple-Robotics/aligator/blob/main/CMakeLists.txt#L137-L141) : creating an explicit EIGENPY_BUILD_WITH_UBUNTU_22_COMPATIBILITY option that must be activated on this system.

ahoarau replied:
Added a `FindCHOLMOD.cmake` in `jrl-v2`. This finder looks for the libraries on ubuntu 22 mimicking the config-based targets in newer versions. If the `CHOLMODConfig.cmake` is available, then the finder does nothing. 

No extra variable needed, easy transition. When Ubuntu 22.04 support is dropped, we can simply remove the `FindCHOLMOD.cmake` in the `jrl-v2`, and `eigenpy` will not need any code modification to read the config directly

jorisv commented on CMakeLists.txt:
```diff
@@ -483,11 +321,21 @@ add_subdirectory(python)
+# ----------------------------------------------------
+jrl_generate_ros2_package_files(
+  INSTALL_CPP_PACKAGE_FILES [[NOT BUILD_STANDALONE_PYTHON_INTERFACE]]
+  INSTALL_PYTHON_PACKAGE_FILES [[BUILD_PYTHON_INTERFACE]]
```
```suggestion
  INSTALL_CPP_PACKAGE_FILES ON
  INSTALL_PYTHON_PACKAGE_FILES ON
```
There is no `BUILD_STANDALONE_PYTHON_INTERFACE` and `BUILD_PYTHON_INTERFACE`

jorisv commented on unittest/python/CMakeLists.txt:
```diff
@@ -0,0 +1,98 @@
+    set(script_path ${CMAKE_CURRENT_SOURCE_DIR}/test_${name}.py)
+  endif()
+
+  set(test_name "Eigenpy: test_${name}.py")
```
```suggestion
  set(test_name "eigenpy: test_${name}.py")
```
for consistency with cpp test and other project tests

jorisv commented on .github/workflows/jrl-cmakemodules.yml:
```diff
@@ -5,46 +5,62 @@ on:
-      - run: sudo apt install libboost-all-dev libeigen3-dev python3-numpy python3-scipy
-      - run: cmake -B build -S eigenpy
-      - run: grep -qvz CMAKE_PROJECT_VERSION:STATIC=0.0 build/CMakeCache.txt
+          repository: ahoarau/jrl-cmakemodules
```
Update after jrl-next release

jorisv commented on .github/workflows/linux.yml:
```diff
@@ -1,63 +1,90 @@
       matrix:
-        python: [3]
-        ubuntu: [22, 24]
+        os: [ubuntu-22.04, ubuntu-24.04]
```
Use docker container `ubuntu:22.04` and `ubuntu:24.04' instead of Github image to have right package version.

jorisv commented on .github/workflows/linux.yml:
```diff
@@ -1,63 +1,90 @@
+          restore-keys: ccache-${{ matrix.os }}-${{ matrix.build_type }}-
+
+      - name: Update APT repositories
+        run: sudo apt-get update
```
You can remove all `sudo` when running on docker images

jorisv commented on .github/workflows/macos-linux-windows-pixi.yml:
```diff
@@ -64,35 +64,41 @@ jobs:
-        key: ccache-macos-linux-windows-pixi-${{ matrix.os }}-${{ matrix.build_type }}-${{ matrix.cxx_options }}-${{ matrix.environment }}-${{ github.sha }}
-        restore-keys: ccache-macos-linux-windows-pixi-${{ matrix.os }}-${{ matrix.build_type }}-${{ matrix.cxx_options }}-${{ matrix.environment }}-
+        path: ${{ env.CCACHE_DIR }}
+        key: ccache-${{ matrix.os }}-${{ matrix.build_type }}-${{ github.sha }}
```
You need to keep cxx_options and environment.
If you don't, a build with `all-python-oldest` environment can retrieve cache from `all` that is not compatible.

You also need to keep`ccache-macos-linux-windows-pixi-` prefix because cache can be shared by [workflow](https://docs.github.com/en/actions/reference/workflows-and-actions/dependency-caching#restrictions-for-accessing-a-cache):
> Multiple workflow runs in a repository can share caches. A cache created for a branch in a workflow run can be accessed and restored from another workflow run for the same repository and branch.

jorisv commented on .github/workflows/linux.yml:
```diff
@@ -1,63 +1,90 @@
-      - run: make -j2
-      - run: ctest --output-on-failure
+          path: ${{ env.CCACHE_DIR }}
+          key: ccache-${{ matrix.os }}-${{ matrix.build_type }}-${{ github.sha }}
```
Find an unique prefix (cf github cache discussion in macos-linux-windows-pixi workflow)

jorisv commented on .github/workflows/macos-linux-windows-pixi.yml:
```diff
@@ -64,35 +64,41 @@ jobs:
+      env:
+        CMAKE_BUILD_PARALLEL_LEVEL: 2
+      run: |
+        pixi run -e ${{ matrix.environment }} build --verbose --parallel 2
```
By running `pixi run build` you will rerun `pixi run configure`.
Pixi can't detect this first step is already done.

ahoarau replied:
added `--skip-deps`

jorisv commented on cmake/get-jrl-cmakemodules.cmake:
```diff
@@ -0,0 +1,49 @@
+endif()
+
+# Fallback to FetchContent if not found
+set(JRL_GIT_REPOSITORY "https://github.com/ahoarau/jrl-cmakemodules.git")
```
To update after jrl-next release

jorisv commented on pixi.toml:
```diff
@@ -15,11 +15,11 @@ cxx-compiler = ">=1.7.0"
 numpy = ">=1.22.0"
-python = ">=3.9.0"
+python = ">=3.10"
 scipy = ">=1.10.0"
```
Add `jrl-cmakemodules = ">=2.0"` when jrl-next is released

jorisv commented on pixi.toml:
```diff
@@ -48,19 +47,19 @@ extra-args = [
 pkg-config = ">=0.29.2"
```
Remove pkg-config and add `jrl-cmakemodules = ">=2.0"` when jrl-next is released

jorisv commented on unittest/python/CMakeLists.txt:
```diff
@@ -0,0 +1,98 @@
+  add_test(NAME ${test_name} COMMAND $<TARGET_FILE:Python::Interpreter> ${script_path})
+  set_tests_properties(
+    ${test_name}
+    PROPERTIES ENVIRONMENT PYTHONPATH=$<TARGET_FILE_DIR:eigenpy_pywrap>/.. DEPENDS eigenpy_pywrap
```
It's better to use ENVIRONMENT_MODIFICATION to avoid issue with some package manager or custom installation path:
```
      ENVIRONMENT_MODIFICATION
        "PYTHONPATH=path_list_prepend:$<TARGET_FILE_DIR:eigenpy_pywrap>/.."
```

ahoarau committed:
cmake: use FindCHOLMOD.cmake provided by jrl-v2
This find module is a backport for Ubuntu 22, and simply uses the config from more recent distros

ahoarau committed:
ci: update ros-ci

ahoarau committed:
cmake: always generate ros files (c++ and py)

ahoarau committed:
tests: Eigenpy -> eigenpy

ahoarau committed:
ci-pixi: add unique name + pur back options and env for cache

ahoarau committed:
ci-pixi: add --skip-deps to avoid task duplication

ahoarau committed:
pixi: drop pkg-config dep

ahoarau committed:
pre-commit: upgrade to latest version

ahoarau committed:
tests: use ENVIRONMENT_MODIFICATION

ahoarau committed:
ci-ubuntu: use docker images

ahoarau committed:
pixi: update lock file

ahoarau committed:
pre-commit: tmp downgrade ruff to 0.15 to avoid breaks

ahoarau committed:
cmake: enforce boost python3

ahoarau committed:
ros-ci: add lyrical

ahoarau committed:
pixi: add jrl-cmakemodules v2 dep

ahoarau committed:
gitignore: ignore .conda files

ahoarau committed:
pixi: use latest pixi build version

ahoarau committed:
jrl: update link to jrl-umi3218 (upstream)

ahoarau committed:
cmake: DEPENDS is reserved for test names

ahoarau committed:
cmake: add EXPECTED_TARGETS

jorisv commented on include/eigenpy/fwd.hpp:
```diff
@@ -25,8 +25,6 @@
 #define EIGENPY_WITH_CXX11_SUPPORT
 #endif
 
-#define EIGENPY_STRING_LITERAL(string) #string
-#define EIGENPY_STRINGIZE(string) EIGENPY_STRING_LITERAL(string)
```
Why remove this ?

ahoarau replied:
<img width="422" height="97" alt="image" src="https://github.com/user-attachments/assets/21f86a13-c408-4ffa-8214-874893eb3929" />

jorisv commented on unittest/python/CMakeLists.txt:
```diff
@@ -0,0 +1,99 @@
+    set(script_path ${CMAKE_CURRENT_SOURCE_DIR}/test_${name}.py)
+  endif()
+
+  set(test_name "eigenpy: test_${name}.py")
```
```suggestion
  set(test_name "eigenpy-test-py-bind-${name}.py")
```

jorisv commented on unittest/cpp/CMakeLists.txt:
```diff
@@ -0,0 +1,85 @@
+  )
+
+  add_test(
+    NAME "eigenpy: import ${name}"
```
```suggestion
    NAME "eigenpy-test-cpp-import-${name}"
```

jorisv commented on unittest/cpp/CMakeLists.txt:
```diff
@@ -0,0 +1,85 @@
+    COMMAND $<TARGET_FILE:Python::Interpreter> -c "import ${name}"
+  )
+
+  set(test_name "eigenpy: test_${name}.py")
```
```suggestion
  set(test_name "eigenpy-test-cpp-${name}")
```

jorisv commented on CMakeLists.txt:
```diff
@@ -1,479 +1,317 @@
+  "Generate the Python stubs associated to the Python library"
+  OFF
+)
 
```
It miss BUILD_DOCUMENTATION and INSTALL_DOCUMENTATION option

jorisv commented on CMakeLists.txt:
```diff
@@ -1,479 +1,317 @@
-# If needed, fix CMake policy for APPLE systems
-apply_default_apple_configuration()
-check_minimal_cxx_standard(11 ENFORCE)
+jrl_option(BUILD_WITH_ACCELERATE_SUPPORT
+           "Build EigenPy with the Accelerate support (Apple only)" OFF
+)
```
Can we add a condition on APPLE ?

jorisv commented on CMakeLists.txt:
```diff
@@ -1,479 +1,317 @@
-    include/eigenpy/utils/python-compat.hpp
-    include/eigenpy/utils/empty-visitor.hpp)
+set(
+  eigenpy_HEADERS
```
Use `jrl_target_headers` and `target_sources` instead of list.

jorisv commented on CMakeLists.txt:
```diff
@@ -1,479 +1,317 @@
+
+add_library(eigenpy SHARED ${eigenpy_SOURCES} ${eigenpy_HEADERS})
+add_library(eigenpy::eigenpy ALIAS eigenpy)
+jrl_target_enforce_msvc_conformance(eigenpy PRIVATE)
```
Same discussion than other project : Do we have to set this to PUBLIC ?

jorisv replied:
You forget to call jrl_target_set_default_compile_options

jorisv commented on CMakeLists.txt:
```diff
@@ -1,479 +1,317 @@
+    Python::Module
+    Python::NumPy
+    Boost::python
+    Boost::numpy
```
Are you sure of Boost::numpy link ? I don't find it it the legacy cmakelists and I don't see any include of boost/python/numpy/ headers

jorisv commented on CMakeLists.txt:
```diff
@@ -1,479 +1,317 @@
 
-option(BUILD_WITH_CHOLMOD_SUPPORT "Build EigenPy with the Cholmod support" OFF)
+jrl_configure_defaults()
 
```
It miss SUFFIX_SO_VERSION option (used bellow)

ahoarau committed:
cmake: rename tests names (revert)

ahoarau committed:
build the documentation via doxygen

ahoarau committed:
cmake: add APPLE condition on BUILD_WITH_ACCELERATE_SUPPORT

ahoarau committed:
cmake: add SUFFIX_SO_VERSION option

ahoarau committed:
cmake: make jrl_target_enforce_msvc_conformance PUBLIC

ahoarau committed:
cmake: add jrl_target_set_default_compile_options

ahoarau committed:
add missing doxygen+graphviz deps

ahoarau committed:
cmake: remove useless Boost::numpy

jorisv committed:
doc: Prefix doc target by eigenpy-

jorisv committed:
pixi: Create new-version environment

jorisv committed:
dev: Add release instruction

ahoarau committed:
cmake: use target_sources

ahoarau committed:
BUILD_DOCUMENTATION is now OFF by default
We now require doxygen+graphviz if set to ON.

ahoarau committed:
cmake: install pywrap as TARGET for rpath
